src/kvstore/Part.cpp

```diff
@@ -5,34 +5,235 @@
  */
 
 #include "kvstore/Part.h"
+#include "wal/BufferFlusher.h"
+#include "kvstore/LogEncoder.h"
+
+DEFINE_int32(cluster_id, 0, "A unique id for each cluster");
 
 namespace nebula {
 namespace kvstore {
 
-void SimplePart::asyncMultiPut(std::vector<KV> keyValues, KVCallback cb) {
-    CHECK_NOTNULL(engine_);
-    auto ret = engine_->multiPut(std::move(keyValues));
-    cb(ret, HostAddr(0, 0));
+using raftex::AppendLogResult;
+
+const char kLastCommittedIdKey[] = "_last_committed_log_id";
+
+namespace {
+
+ResultCode toResultCode(AppendLogResult res) {
+    switch (res) {
+        case AppendLogResult::SUCCEEDED:
+            return ResultCode::SUCCEEDED;
+        case AppendLogResult::E_NOT_A_LEADER:
+            return ResultCode::ERR_LEADER_CHANGED;
+        default:
+            return ResultCode::ERR_CONSENSUS_ERROR;
+    }
 }
 
-void SimplePart::asyncRemove(const std::string& key, KVCallback cb) {
-    CHECK_NOTNULL(engine_);
-    auto ret = engine_->remove(key);
-    cb(ret, HostAddr(0, 0));
+
+wal::BufferFlusher* getBufferFlusher() {
+    static wal::BufferFlusher flusher;
```

Comment:
Only one flusher maybe a bottleneck.  Let's open another issue to make the number configurable.

Reply:
We could make the number of flushing threads in the BufferFlusher configurable

Sure. let's do it in a separate PR